src/controllers/userProfileController.ts

```diff
@@ -0,0 +1,42 @@
+import { Request, Response } from 'express'
+import User from '../database/models/userModel'
+import { profileValidationalSchema } from '../validations/userValidation'
+
+export const updateUser = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const fieldsToUpdate = req.body
+    const { error } = profileValidationalSchema.validate(fieldsToUpdate)
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message })
+    }
+    if (Object.keys(fieldsToUpdate).length === 0) {
+      return res.status(400).json({ message: 'nothing to update' })
+    }
+
+    const user = await User.findOne({ where: { id } })
+    const result = await User.update(fieldsToUpdate, {
```

Comment:
```suggestion
```
remember to add proper check for updating email or password, updating email requires email verification